sim/warrior/heroic_leap.go

```diff
@@ -0,0 +1,50 @@
+package warrior
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+)
+
+func (warrior *Warrior) RegisterHeroicLeap() {
+
+	numHits := warrior.Env.GetNumTargets()
+	warrior.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 6544},
+		SpellSchool:    core.SpellSchoolPhysical,
+		ProcMask:       core.ProcMaskMeleeMHSpecial,
+		Flags:          core.SpellFlagMeleeMetrics | core.SpellFlagAPL,
+		ClassSpellMask: SpellMaskHeroicLeap | SpellMaskSpecialAttack,
```

Comment:
The class spell mask needs to unique and right now you're giving every single special attack bit to it. Leave only the HeroicLeap spell mask here and add the heroic leap to the specialattack masks if that is what you want to do

Reply:
Okay, Most of the other Warrior spells are implemented this way.  If you prefer I can add to special attack

Reply:
Ohh sorry I see now how warriors implements the class masks. Weird but I guess it was more convenient this way. Sorry you can leave this as it is